File: Algorithms_Sorting_CPP/SelectionSort/SelectionSort.cpp
#include <iostream>

static void runSelectionSort(int*, int);
static void displayList(int[], int);

int main() {
	int numbers[] = {900, 32, 75, 84, 35, 33, 21, 8, 9, 2, 1};
	int size = sizeof(numbers) / sizeof(numbers[0]);
	std::cout << "Unsorted List: ";
	displayList(numbers, size);
	runSelectionSort(numbers, size);
	std::cout << "Sorted List: ";
	displayList(numbers, size);
}

static void runSelectionSort(int* numbersArray, int size) {
	
	for (int i = 0; i < size - 1; i++) {
		int minIndex = i;
		for (int j = i + 1; j < size; j++) {
			if (numbersArray[j] < numbersArray[minIndex]) {
				minIndex = j;
			}
		}
		
		int temp = numbersArray[minIndex];
		numbersArray[minIndex] = numbersArray[i];
		numbersArray[i] = temp;
	}
}

static void displayList(int numbers[], int size) {
	int moves = 0;

	std::cout << "[";
	for (int i = 0; i < size; i++) {
		if (moves == size - 1)
			std::cout << numbers[i];
		else {
			std::cout << numbers[i] << ", ";
			moves++;
		}
	}
	std::cout << "]\n";
}

File: algorithms/sorting/selection-sort/selection-sort-implementations/java/SelectionSort/src/main/java/demos/java/core/algorithms/sorting/SelectionSort.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Project/Maven2/JavaApp/src/main/java/${packagePath}/${mainClassName}.java to edit this template
 */

package demos.java.core.algorithms.sorting;

/**
 *
 * @author jdcodes-12
 */
public class SelectionSort {

    public static void main(String[] args) {
      int[] numbers = { 101, 30, 9, 47, 62, 65, 89, 700, 640};
      System.out.print("Unsorted list: ");
      displayList(numbers);
      runSelectionSort(numbers);
      System.out.print("Sorted list: ");
      displayList(numbers);
    }
   
    private static void runSelectionSort(int[] numbers) {
        int n = numbers.length;
        int minIndex, i, j, temp;
        
        for (i = 0; i < n; i++){
            minIndex = i;
            for (j = i+1; j < n; j++){
                if (numbers[j] < numbers[minIndex]) {
                    minIndex = j;
                }
            }
            
            temp = numbers[minIndex];
            numbers[minIndex] = numbers[i];
            numbers[i] = temp;
        }
    }
 
    private static void displayList(int[] numbers) {
    
        int moves = 0;
        
        System.out.print("[");
        for (int number : numbers) {
            if (moves == numbers.length - 1) 
                System.out.print(number);
            else {
                System.out.print(number + ", ");
                moves++;
            }
        }
        System.out.println("]");
    }
}
